pkg/util/k8s.go

```diff
@@ -0,0 +1,31 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+)
+
+// EncodeKubernetesName takes a string and turns it into a form suitable for use as a Kubernetes name.
+// K8S names are lower case, numbers, '-', and '.'. Invalid characters are percent-encoded in a _similar_ way to URL
+// encoding, with a percent followed by the numerical character code. Conversion is:
+//   Upper case letters -> lower case.
+//   'a-z', '0-9', '-', '.' -> left as-is.
+//   Any other characters: percent-encoded ('%' + rune hex code).
+func EncodeKubernetesName(name string) string {
+	name = strings.ToLower(name)
+	encodedName := strings.Builder{}
+
+	for _, ch := range name {
+		if allowedInK8SName(ch) {
+			encodedName.WriteRune(ch)
+		} else {
+			encodedName.WriteString(fmt.Sprintf("%%%X", int(ch)))
```

Comment:
do we need a function to "UnEncode" these (e.g. jx get builds) ?  what does the tools show (hopefully not the percent encoded version?)

Reply:
Not that I can see atm.
tbh, I'm not entirely sure where we display these atm, but yes tools could well show them. For the values that currently work, the tools will show the same stuff. For the values that currently don't work at all, the tools should display some weird stuff. But at least it'll work. :-)